src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,108 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.RegistrationException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationServiceImpl registrationService;
+    private static StorageDaoImpl storage;
+    private static final String VALID_LOGIN_1 = "validLogin1";
+    private static final String VALID_LOGIN_2 = "validLogin2";
+    private static final String EMPTY_LOGIN = "";
+    private static final String VALID_PASSWORD = "validPassword";
+    private static final String EMPTY_PASSWORD = "";
+    private static final String INVALID_LENGTH = "short";
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+        storage = new StorageDaoImpl();
+
+    }
+
+    @Test
+    void validRegistration_Ok() {
+        User validUser = new User(VALID_LOGIN_1, VALID_PASSWORD, 20);
+        assertDoesNotThrow(() -> registrationService.register(validUser));
+    }
+
+    @Test
+    void duplicateLogin_NotOk() {
+        User user1 = new User(VALID_LOGIN_2, VALID_PASSWORD, 25);
+        User user2 = new User(VALID_LOGIN_2, VALID_PASSWORD, 30);
+
+        assertDoesNotThrow(() -> registrationService.register(user1));
+        assertThrows(RegistrationException.class,
+                () -> registrationService.register(user2));
```

Comment:
we can't rely on the method we test to test said method
![image](https://github.com/mate-academy/jv-registration-validation/assets/49767049/22069c84-3d79-4d68-84d0-618d8ece505d)